### ISSUE TITLE
Crash while trying to change directory while saving

### ISSUE THREAD

**Issue by MuTaTeD** [xournalpp/xournalpp#999]
The application crashes when trying to change directory while trying to save the file., even with a blank file. Have to restore it from autosave and save in the ~/.xournalpp/ directory and then manually copy to the required directory. Also experienced the same on previous versions

Version: V1.0.9 Buillt: Mar 20 2019
OS: Kubuntu 18.04 LTS 

Let me know if some other information is required.

CrashLog is attached
[errorlog.20190320-161301.log](https://github.com/xournalpp/xournalpp/files/2987878/errorlog.20190320-161301.log)

**Comment by LittleHuba** [comment]
Would you mind creating a video so I can triage the bug? The error log does unfortunately not help very much.

**Comment by MuTaTeD** [comment]
The video is uploaded at 
https://www.dropbox.com/s/aax5h9o9tmu70xv/Screencast%202019-03-25%2023%3A05%3A15.mp4?dl=1

**Comment by LittleHuba** [comment]
Can you try the file picker in the `gtk-demo`? It is in the `Pickers` demo. Does that crash as well?

**Comment by MuTaTeD** [comment]
all the gtk-demo pickers work correctly, do you want me to try gtk3-demos as well

**Comment by LittleHuba** [comment]
Yes please try the `gtk3-demo` as well.

**Comment by MuTaTeD** [comment]
Pickers in the gtk3-demo are also working correctly...

**Comment by LittleHuba** [comment]
Can you rebuild Xournal++ and check whether that solves the issue? The save dialog is something like a blackbox that is just called by us. The error happens somewhere within the dialog when the layout is done/changed. Best guess is that you updated GTK after installing Xournal++ and that broke something (what should actually not be the case).

**Comment by MuTaTeD** [comment]
I built Xournal++ on my machine and had it crash successfully
The output on the console was
```
Gtk-Message: 17:19:12.517: GtkDialog mapped without a transient parent. This is discouraged.
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Icon 'image-missing' not present in theme breeze (gtk-icon-theme-error-quark, 0)

** (xournalpp:18178): WARNING **: 17:19:20.793: [Crash Handler] Crashed with signal 6

** (xournalpp:18178): WARNING **: 17:19:20.795: [Crash Handler] Wrote crash log to: /home/mutated/.xournalpp/errorlogs/errorlog.20190329-171920.log

** (xournalpp:18178): WARNING **: 17:19:21.249: Trying to emergency save the current open document…

** (xournalpp:18178): WARNING **: 17:19:21.249: Successfully saved document to "/home/mutated/.xournalpp/emergencysave.xopp"
```

**Comment by LittleHuba** [comment]
Ahh that error message has a lot more details. 
It seems your theme is missing a crucial icon required by the dialog. As we are not using anything special in the dialog it is most likely something in the directory you traverse to. You could test this by changing the theme and trying it again. 

The error is not in Xournal++ but in the theme you are using.

**Comment by MuTaTeD** [comment]
Got It working finally, changed the GTK icon theme from Breeze to Oxygen and it is working now
For future reference: `System Settings -> Application Style -> Gnome Application Style (GTK)`

**Comment by ethpi** [comment]
Works for me with Kubuntu 18.04, if I change the **symbol-design** of the Gnome Application Style. Thanks.

But in Kubuntu 20.04 I can't find a button to select the symbol-design in System Settings. I only can select the Gnome Application Style. But this change doesn't solve the problem. Any ideas how not to use the Breeze symbols but the Oxygen symbols in Kubuntu 20.04?

**Comment by ethpi** [comment]
interrmediate result:
In Plasma 5 shipped with Kubuntu 20.04 you can change the icon set for all applications in System Settings -> Symbols. If I use another icon set than breeze xournal++ does not crash.

**Comment by Pointedstick** [comment]
We (KDE developers) can add an `image-missing` icon to the Breeze icon theme. However it's undoubtedly a bug in the app or in GTK that it crashes when this icon is missing. At worst, it should display nothing. Missing icons should never crash an app, that's just crazy!
